server/test/integration/vitalsTest/checkForInternalProblemsTest.js

```diff
@@ -0,0 +1,420 @@
+        expect(helpers.logSentry).to.have.been.calledWithExactly(
+          `Unusually frequent number of stillness alerts (${maxStillnessAlerts + 1}) have been received at location2`,
+        )
+      })
+
+      it('should send one max stillness alert message specific to the third location that has exceeded the maxStillnessAlerts in set interval of time to Sentry', () => {
+        expect(helpers.logSentry).to.have.been.calledWithExactly(
+          `Unusually frequent number of stillness alerts (${maxStillnessAlerts + 1}) have been received at location3`,
+        )
+      })
+
+      it('should send a total of three alerts (since there are three locations)', () => {
+        expect(helpers.logSentry).to.have.been.calledThrice
+      })
+
+      it('should not log any errors', () => {
+        expect(helpers.log).to.not.be.called
+      })
+    })
+    describe('only one location exceeds maxStillnessAlerts in set interval of time, whereas other locations have exactly max and less than', () => {
+      beforeEach(async () => {
+        // Insert a single location and maxStillnessAlerts sessions + 1
+        const client = await factories.clientDBFactory(db, { isSendingAlerts: true })
+        for (let i = 1; i <= 3; i += 1) {
+          this[`testLocation${i}`] = await locationDBFactory(db, {
+            clientId: client.id,
+            locationid: `location${i}`,
+            isSendingAlerts: true,
+          })
+        }
+
+        this.incidentCategory = ALERT_TYPE.SENSOR_STILLNESS
+        this.alertType = ALERT_TYPE.SENSOR_STILLNESS
+        for (let i = 1; i <= maxStillnessAlerts; i += 1) {
+          await sessionDBFactory(db, {
+            locationid: this.testLocation1.locationid,
+            alertType: this.alertType,
+            incidentCategory: this.incidentCategory,
+          })
+        }
+
+        for (let i = 1; i <= maxStillnessAlerts + 1; i += 1) {
+          await sessionDBFactory(db, {
+            locationid: this.testLocation2.locationid,
+            alertType: this.alertType,
+            incidentCategory: this.incidentCategory,
+          })
+        }
+
+        for (let i = 1; i <= maxStillnessAlerts + 1; i += 1) {
+          await sessionDBFactory(db, {
+            locationid: this.testLocation3.locationid,
+            alertType: ALERT_TYPE.SENSOR_DURATION,
+            incidentCategory: ALERT_TYPE.SENSOR_DURATION,
+          })
+        }
+        await vitals.checkForInternalProblems()
+      })
+
+      it('should only send one max stillness alert message specific to the second location that has exceeded the maxStillnessAlerts in set interval of time to Sentry', () => {
+        expect(helpers.logSentry).to.have.been.calledOnceWithExactly(
+          `Unusually frequent number of stillness alerts (${maxStillnessAlerts + 1}) have been received at location2`,
+        )
+      })
+
+      it('should not log any errors', () => {
+        expect(helpers.log).to.not.be.called
+      })
+    })
+  })
+})
```

Comment:
This whole thing is great. Very thorough testing. Thank you for doing all this :smile: